apps/sms/views/conversation/js/attachment.js

```diff
@@ -26,8 +26,7 @@
   function Attachment(blob, options) {
     options = options || {};
     this.blob = blob;
-    this.name = blob.name || options.name ||
-      navigator.mozL10n.get('unnamed-attachment');
+    this.name = blob.name || options.name || 'undefined';
```

Comment:
from what I read in the code (basically `git grep 'new Attachment`) in most cases the name comes from external sources (activities). So we're never sure we'll have it properly. So this can definitely happen in some cases and we need to localize it.

Currently, it's used in these locations:
* https://github.com/mozilla-b2g/gaia/blob/956982413ef87c192ebc778012f6c085e678570b/apps/sms/views/conversation/js/conversation.js#L51 (this is then used when generating the SMIL markup when we send the MMS -- happens only after a user action)
* https://github.com/mozilla-b2g/gaia/blob/956982413ef87c192ebc778012f6c085e678570b/apps/sms/views/conversation/js/compose.js#L750 (used when displaying the file in gallery -- happens only after a user action)
* https://github.com/mozilla-b2g/gaia/blob/956982413ef87c192ebc778012f6c085e678570b/apps/sms/views/conversation/js/attachment_renderer.js#L278 (used when displaying the file in the app itself -- it's shown only when it's not an image). Note that this is called from `attachment.render()` so we can pretty well control when it's called.
* another place in attachment.js but it won't matter if we localize it or not in this place.

So here is my idea to make it work:
```javascript
function Attachment(...) {
  ...
  this.name = blob.name || options.name;
  if (this.name) {
    this.namePromise = Promise.resolve(this.name);
  } else {
    this.name = 'untitled';
    this.namePromise = navigator.mozL10n.formatValue('unnamed-attachment').then(
      (val) => this.name = val
    );
  }
  ...
}

...
render: function(readyCallback) {
  var attachmentRenderer = this._getAttachmentRenderer();

  this.namePromise.then(() => attachmentRenderer.render()).catch(function(e) {
    console.error('Error occurred while rendering attachment.', e);
  }).then(readyCallback);

  // We still need this for the case where we render a list of attachments
  // and right order is important.
  return attachmentRenderer.getAttachmentContainer();
},
```

I think all cases are covered with this?